ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -584,6 +578,7 @@ private int executeIncrementalLoad() {
         DAGTraversal.traverse(childTasks, new AddDependencyToLeaves(TaskFactory.get(work, conf)));
       }
       this.childTasks = childTasks;
+      createReplLoadCompleteAckTask();
```

Comment:
This should be done only once when no more tasks are there to execute .. "if (builder.hasMoreWork() || work.getPathsToCopyIterator().hasNext() || work.hasBootstrapLoadTasks()) {'

Reply:
The check is inside the method